apps/admin/src/actions/user.ts

```diff
@@ -98,19 +111,151 @@ export async function updateUser(data: z.infer<typeof baseInformationSchema>, id
 
 export async function updateUserState(data: z.infer<typeof userStateSchema>, id: string) {
     await checkIfAdmin();
+    const userData = await getUserById(id);
+    if (!userData) {
+        throw new Error("User not found");
+    }
+
+    if (data.experimentStatus === "installation" && !data.installationDate) {
+        throw new Error("Installation date is required for installation status");
+    }
+
+    if (data.experimentStatus === "deinstallation" && !data.deinstallationDate) {
+        throw new Error("Deinstallation date is required for deinstallation status");
+    }
 
     try {
+        let activeDate: Date | null = null;
+        if (!userData.isActive && data.isActive) {
+            activeDate = new Date();
+            activeDate.setHours(0, 0, 0, 0);
+        }
         await updateUserDb(
             {
+                isActive: data.isActive,
                 isAdmin: data.isAdmin,
-                isActive: data.active,
                 isParticipant: data.isParticipant,
                 appVersion: data.appVersion,
+                activationDate: activeDate,
             },
             id,
         );
-        revalidatePath("/users");
     } catch (e) {
         throw new Error("Failed to update user");
     }
+
+    if (!userData.isActive && data.isActive) {
+        try {
+            await sendAccountActivatedEmail({
+                to: userData.email,
+                name: userData.username,
+                from: env.RESEND_API_MAIL,
+                apiKey: env.RESEND_API_KEY,
+            });
+        } catch (err) {
+            console.error("Failed to send account activated mail", err);
```

Comment:
Wollen wir tatsächlich fehlerhaft gesendete Mails vllt auch in unsere Logs einfügen und nicht in die console?